### PR TITLE
Manage database models and migration with Prisma

### LINKED ISSUE #3415: Database migration with Prisma
We were talking in slack about how to handle our Postgres database migration.
Let's explore how to integrate Prisma to define database schema and manage migrations.

Resources:

https://www.prisma.io/docs/concepts/components/prisma-migrate
https://supabase.com/docs/guides/integrations/prisma

### PR DESCRIPTION
**Pull request by DukeManh**
### Why

Currently, we define our database models in [04-telescope-schema.sql](https://github.com/Seneca-CDOT/telescope/blob/master/docker/supabase/volumes/db/init/04-telescope-schema.sql), and [05-github-schema.sql](https://github.com/Seneca-CDOT/telescope/blob/master/docker/supabase/volumes/db/init/05-github-schema.sql). That is not the right approach, because when we change existing models or add new models, the current database will not be updated because these scripts only run during database initialization.

When managing a database, we often use an [ORM](https://blog.bitsrc.io/what-is-an-orm-and-why-you-should-use-it-b2b6f75f5e2a) to manipulate our database schemas in our language of choice and to trigger database migration when there're schema changes.

> Object-relational-mapping is the idea of being able to write queries using the object-oriented paradigm of your preferred programming language.

> Database migrations basically track granular changes to your database schema (and sometimes to your data as well). These granular changes are typically reflected as separate scripted files. That way, your granular schema changes are reflected as code that can be captured with any version control software. 

### How: 

Come [Prisma](https://www.prisma.io/), the hottest ORM tool for NodeJS.

With Prisma, data models are defined in `schema.prisma`
```
model feeds {
  url                String              @id
  user_id            String? 
```

We currently have defined some Models, to generate Prisma Models for existing db, run:

```
 cd src/db
 pnpx prisma db pull
```

Apply database migration:
```
 pnpx prisma migrate dev
```
Sometimes we want to do custom db migration, i.e, add RLS rules to the tables.
1. Create migration file
```
 pnpx prisma migrate dev --create-only
```
2. Edit the newly created `migration.sql` file
3. Apply migration
```
pnpx prisma migrate dev
```
Applied migrations files should not be edited and must be committed to source control to keep track of database transformation.

## Issue This PR Addresses
fix #3415

### CODE REVIEW

**Comment by dbelokon** [comment]
Needs to  have a meeting for more discussion. Talk on slack

File: .github/workflows/e2e-tests-ci.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
     env:
       # Whichever Docker containers are needed to pass e2e tests, define them here
       DOCKER_CONTAINERS: 'sso feed-discovery auth login kong rest meta elasticsearch redis posts traefik'
+      # Postgres database connection string to run db migration
```
[the remark this answers is not in the document]
Comment (DukeManh) [reply .github/workflows/e2e-tests-ci.yml:16]: It worked!

[the remark this answers is not in the document]
Comment (DukeManh) [reply .github/workflows/e2e-tests-ci.yml:16]: https://github.com/Seneca-CDOT/telescope/runs/5877151144?check_suite_focus=true

File: src/db/prisma/migrations/20220406232746_change_feeeds_type_data_type/migration.sql [hunk 2]
```diff
@@ -0,0 +1,95 @@
+-- Case-insensitive text type
```
Comment (humphd) [inline src/db/prisma/migrations/20220406232746_change_feeeds_type_data_type/migration.sql:1]: 3 `eee`s on the filename `feeeds`

File: src/db/prisma/migrations/migration_lock.toml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Please do not edit this file manually
+# It should be added in your version-control system (i.e. Git)
+provider = "postgresql"
```
Comment (humphd) [inline src/db/prisma/migrations/migration_lock.toml:3]: Newline at end please

**Review by humphd — CHANGES_REQUESTED** [review]
Can you throw a README in `src/db` to explain what it is, and how/when to use it?

**Comment by humphd** [comment]
Filed https://github.com/Seneca-CDOT/telescope/issues/3435 for after this release.

File: src/db/prisma/migrations/migration_lock.toml (hunk 3 above)
Comment (humphd) [reply src/db/prisma/migrations/migration_lock.toml:3]: So fix it upstream :) https://github.com/prisma/prisma-engines/blob/45e6871398e3e66d16be7a12a029938abdc1709c/migration-engine/connectors/migration-connector/src/migrations_directory.rs#L53-L74

File: src/db/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "version": "1.0.0",
+  "description": "Managing Postgres database migration",
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "prisma": "3.12.0"
+  },
+  "scripts": {
+    "migrate": "prisma migrate dev",
```
Comment (humphd) [inline src/db/package.json:12]: Question: what does the `dev` refer to?

**Review by humphd — APPROVED** [review]

File: src/db/package.json (hunk 4 above)
Comment (DukeManh) [reply src/db/package.json:12]: Migration on development

**Review by tcvan0707 — APPROVED** [review]

File: src/db/package.json (hunk 4 above)
Comment (humphd) [reply src/db/package.json:12]: In production, how would it differ?

File: src/db/prisma/migrations/migration_lock.toml (hunk 3 above)
Comment (DukeManh) [reply src/db/prisma/migrations/migration_lock.toml:3]: 100%

**Review by JerryHue — APPROVED** [review]

**Pull request merged**